### PR TITLE
Introduce `get_kedro_project_json_data` function for VSCode integration 

### PR DESCRIPTION
**Pull request by jitu5**
## Description

Resolves https://github.com/kedro-org/kedro-viz/issues/2021

To get current Kedro project JSON data required by Kedro-Viz React component when running inside VSCode, We introduce `get_kedro_project_json_data` function in `responses.py` which returns current Kedro project JSON data.   

## Development notes

- This function will call from Kedro VSCode extension 
- Before calling `get_kedro_project_json_data` function, `load_and_populate_data` will be called 

## Checklist

- [X] Read the [contributing](/CONTRIBUTING.md) guidelines
- [X] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by noklam — APPROVED** [review]

**Comment by astrojuanlu** [comment]
And also, would it make sense longer term to extract that functionality to a self-contained package? Thinking for example of the defunct https://github.com/AlpAribal/kedro-inspect/

**Comment by noklam** [comment]
> And also, would it make sense longer term to extract that functionality to a self-contained package? Thinking for example of the defunct https://github.com/AlpAribal/kedro-inspect/

This definitely make sense, but I would like to see this happen at a core level where `kedro` has a better way of serialise/deserialise a `Pipeline`. Otherwise the current JSON data structure is very much coupled to `kedro-viz`, and I don't see much benefit of making it a separate package itself. Maybe just making `kedro-viz` a leaner package would help. similiar to the `kedro-core` discussion, maybe `kedro-viz[core]` or `kedro-viz[standard]`.


https://nbviewer.org/github/noklam/kedro-viz-lite/blob/main/nbs/10_demo.ipynb, regardless of a separate package /refactoring, I would like to see this can be done in a notebook, where I can easily convert an existing `Pipeline` object. The notebook has shown that `Pipeline` is the only thing that is needed, we don't really need a full Kedro project to generate the flowchart JSON.

File: package/kedro_viz/api/rest/responses.py [hunk 1]
```diff
@@ -400,17 +401,42 @@ def get_package_compatibilities_response(
...
+
+def write_api_response_to_fs(file_path: str, response: Any, remote_fs: Any):
+    """Get encoded responses and writes it to a file"""
+    encoded_response = get_encoded_response(response)
+
     with remote_fs.open(file_path, "wb") as file:
         file.write(encoded_response)
 
 
+def get_kedro_project_json_data():
```
[the remark this answers is not in the document]
Comment (jitu5) [reply package/kedro_viz/api/rest/responses.py:422]: @astrojuanlu Could you please clarify the reasoning behind using `root_filepath: Path` as an argument? I'm not sure I understand its purpose in this context.

[the remark this answers is not in the document]
Comment (astrojuanlu) [reply package/kedro_viz/api/rest/responses.py:422]: Not necessarily in this context, but going forward as a way to make this utility function more flexible. Although I agree with https://github.com/kedro-org/kedro-viz/pull/2049#issuecomment-2321494935

**Review by rashidakanchwala — APPROVED** [review]
LGTM. thanks!

**Pull request merged**